mixer/cmd/build.go

```diff
@@ -199,113 +218,158 @@ var buildFormatNewCmd = &cobra.Command{
+		}
 
-		// Build the +20 update so we don't have to switch tooling in between
+		// Set the format back to old for the actual build update
+		b.State.Mix.Format = oldFormat
+
+		// Build the update content for the +10 build
 		params := builder.UpdateParameters{
-			MinVersion:    ver,
-			Format:        strconv.Itoa(newFormat),
+			MinVersion:    buildFlags.minVersion,
+			Format:        b.State.Mix.Format,
 			Publish:       !buildFlags.noPublish,
 			SkipSigning:   buildFlags.noSigning,
 			SkipFullfiles: buildFlags.skipFullfiles,
 			SkipPacks:     buildFlags.skipPacks,
 		}
-		err = b.BuildUpdate(params)
-		if err != nil {
+		if err = b.BuildUpdate(params); err != nil {
 			failf("Couldn't build update: %s", err)
 		}
 
-		// Copy +20 chroots to +10 so we can build last formatN build with the
-		// same content
-		prevVersion, err := strconv.Atoi(b.MixVer)
-		if err != nil {
-			fail(err)
-		}
-		prevVersion -= 10
-		source := filepath.Join(b.Config.Builder.ServerStateDir, "image", b.MixVer)
-		dest := filepath.Join(b.Config.Builder.ServerStateDir, "image", strconv.Itoa(prevVersion))
-		fmt.Println(" Copying +20 chroots to +10 chroots")
-		if err = helpers.RunCommandSilent("cp", "-al", source, dest); err != nil {
-			failf("Failed to copy +20 chroots to +10: %s\n", err)
+		zeroVer := strconv.Itoa(ver - 20)
+		// Write the +0 version to LAST_VER so that we reference in both +10 and +20 manifests as the 'previous:'
+		if err = ioutil.WriteFile(filepath.Join(b.Config.Builder.ServerStateDir, "image", "LAST_VER"), []byte(zeroVer), 0644); err != nil {
+			failf("Couldn't update LAST_VER file: %s", err)
 		}
+	},
+}
 
-		// Copy the old groups.ini file back which contains ALL original bundle names
-		// to account for any removed bundles in this build when creating manifests
-		fmt.Println(" Copying full groups.ini back to working directory")
-		if err = b.RevertFullGroupsINI(); err != nil {
+// This is the first build in the new format. The content is the same as the +10
+// but the manifests might be created differently if a new manifest template is
+// defined for the new format.
+var buildFormatNewCmd = &cobra.Command{
+	Use:   "new",
+	Short: "Build the +20 version in the new format for the format bump",
+	Long:  `Build the +20 version in the new format for the format bump`,
+	Run: func(cmd *cobra.Command, args []string) {
+		b, err := builder.NewFromConfig(configFile)
+		if err != nil {
 			fail(err)
 		}
 
-		// Set the format back to the previous format version before building the +10 update
-		prevFormat, err := strconv.Atoi(b.State.Mix.Format)
+		// Set mixversion to +20 build now, we only need to build update since we
+		// already have the bundles created
+		ver, err := strconv.Atoi(b.MixVer)
 		if err != nil {
 			fail(err)
 		}
-		prevFormat--
-		if err = b.UpdateFormatVersion(strconv.Itoa(prevFormat)); err != nil {
-			fail(err)
+		if err = b.UpdateMixVer(ver + 10); err != nil {
```

Comment:
You initiate `ver` to `b.MixVer` then immeidately start using it at another value. This is still difficult to follow.

All this logic could really use a new type
```go
type fbVersions struct {
    zeroVer, oldFormatVer, newFormatVer int
    oldFormat, newFormat int
}
```
and use that everywhere.

Reply:
hmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmm